utils/new-design/all.py

```diff
@@ -0,0 +1,194 @@
+import os
+import yaml
+import itertools
+
+class ConfigVariants:
+    arch = ['arm64', 'x86_64']
+    hypervisor = ['none', 'kvm', 'xen']
+    platform = ['qemu', 'firecracker', 'xen']
+    build_tool = ['make', 'kraft']
+    run_tool = ['vmm', 'kraft']
+    bootloader = ['multiboot', 'uefi']
+    pie = [False, True]
+    debug = ['none', 'strace', 'err', 'info', 'debug']
+    fs = ['none', 'initrd', '9pfs']
+    networking = ['none', 'bridge', 'nat', 'tap']
+
+    def _convert_to_dict(i):
+        return {
```

Comment:
```suggestion
        return {
```

could you add an assert for the number of options, to make sure we get an error when we add a new option